datafusion/common/src/config.rs

```diff
@@ -212,11 +219,12 @@ config_namespace! {
         pub enable_ident_normalization: bool, default = true
 
         /// When set to true, SQL parser will normalize options value (convert value to lowercase)
-        pub enable_options_value_normalization: bool, default = true
+        pub enable_options_value_normalization: bool, default = false
 
         /// Configure the SQL dialect used by DataFusion's parser; supported values include: Generic,
         /// MySQL, PostgreSQL, Hive, SQLite, Snowflake, Redshift, MsSQL, ClickHouse, BigQuery, and Ansi.
-        pub dialect: String, default = "generic".to_string()
+        pub dialect: String, default = "generic".to_string() // no need to lowercase because
+        // [`sqlparser::dialect_from_str`] is case-insensitive
```

Comment:
Comment formatting is off after it was broken across the lines
Also, the ``[`...`]`` links probably don't work in ordinary comments (but didn't check that)


```suggestion
        // no need to lowercase because `sqlparser::dialect_from_str` is case-insensitive
        pub dialect: String, default = "generic".to_string() 
```

Reply:
Moved to the new line. You are right, `[...]` isn't validated in comments but I left it as is to keep consistent

<img width="945" alt="RustRover 2024-12-11 18 54 39" src="https://github.com/user-attachments/assets/c54e436d-52aa-42dc-b3e8-4a05e2521a49" />